### DISCUSSION TITLE
curl stuck at "TLSv1.3 (OUT), TLS handshake, Client hello (1)"

### DISCUSSION THREAD

**Discussion by lida2003** [curl/curl#11604] [Q&A] [answered]
I'm using raspberry pi with ubuntu OS. 

It seems  "curl -fsSLv https://get.docker.com -o get-docker.sh"  stucked at "TLSv1.3 (OUT), TLS handshake, Client hello (1)".

Any idea?

```
ubuntu@ubuntu:~$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.3 LTS
Release:        22.04
Codename:       jammy
ubuntu@ubuntu:~$ curl --version
curl 7.81.0 (aarch64-unknown-linux-gnu) libcurl/7.81.0 OpenSSL/3.0.2 zlib/1.2.11 brotli/1.0.9 zstd/1.4.8 libidn2/2.3.2 libpsl/0.21.0 (+libidn2/2.3.2) libssh/0.9.6/openssl/zlib nghttp2/1.43.0 librtmp/2.3 OpenLDAP/2.5.15
Release-Date: 2022-01-05
Protocols: dict file ftp ftps gopher gophers http https imap imaps ldap ldaps mqtt pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS brotli GSS-API HSTS HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz NTLM NTLM_WB PSL SPNEGO SSL TLS-SRP UnixSockets zstd
ubuntu@ubuntu:~$ curl -fsSLv https://get.docker.com -o get-docker.sh
* Uses proxy env variable https_proxy == 'https://192.168.78.150:808'
*   Trying 192.168.78.150:808...
* Connected to (nil) (192.168.78.150) port 808 (#0)
* ALPN, offering http/1.1
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* TLSv1.0 (OUT), TLS header, Certificate Status (22):
} [5 bytes data]
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
} [512 bytes data]
```

I'm sure the command [`curl -fsSLv https://get.docker.com -o get-docker.sh`](https://docs.docker.com/engine/install/debian/#install-using-the-convenience-script) is good. 
And my curl version is [7.81.0](https://github.com/curl/curl/discussions/11542).

I don't have any clue right now.

**Comment by lida2003** [comment]
And if I change to use following commands, it works.

So what's wrong with curl or any configuration that I should change?


```
$ wget https://get.docker.com
$ mv index.html get-docker.sh
$ chmod +x get-docker.sh
```

**Comment by jay** [comment, marked as answer]
> * Uses proxy env variable https_proxy == 'https://192.168.78.150:808'

You have a proxy environment variable https_proxy which is the proxy curl will use for https:// URLs. A common mistake is users assume the proxy specified in https_proxy must be https but the proxy is actually http. Try this:

~~~
https_proxy=http://192.168.78.150:808 curl -fsSLv https://get.docker.com -o get-docker.sh
~~~

**Reply by bagder** [reply]
The working wget command line suggests @jay is right as wget does not support HTTPS proxies...
